sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala

```diff
@@ -2053,6 +2053,34 @@ class ParquetV2FilterSuite extends ParquetFilterSuite {
       }
     }
   }
+
+  test("SPARK-38041: Data filter push down with partition filter") {
+    val data = spark.createDataFrame((1 to 4).map(i => Tuple1(Option(i))))
+    withNestedDataFrame(data).foreach { case (inputDF, colName, resultFun) =>
+      withTempPath { file =>
+
+        val partName = "part"
+        inputDF.withColumn(partName, lit(0))
```

Comment:
Is this a partition column? Seems like a data column to me.

Reply:
Ah, I miss the `.partitionBy(partName)`, I will add it in next commit.